### PR TITLE
Enable helm-shell-prompts in all comint-based modes (SLIME, Geiser)

### PR DESCRIPTION
**Pull request by Ambrevar**
This is awesome! :)

Now I'm realizing however that my previous commit for "helm-shell" was a bit shortsighted: a "helm-comint.el" file would have made more sense.

If you agree, I will rename everything to `helm-comint`.

Besides, we've already got some `comint` related functions in `helm-misc`.  Maybe we could move them to `helm-comint.el` as well.

Thoughts?

### CODE REVIEW

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> This is awesome! :)
>
> Now I'm realizing however that my previous commit for "helm-shell" was a bit shortsighted: a "helm-comint.el" file would have made more sense.
>
> If you agree, I will rename everything to helm-comint.

Ok.

> Besides, we've already got some comint related functions in helm-misc. Maybe we could move them to helm-comint.el as well.

Yes, agreed.

> Thoughts?

please don't use `seq` in helm.

> ------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/emacs-helm/helm/pull/2133
>
> Commit Summary
>
>   * Enable helm-shell-prompts in all comint-based modes (SLIME, Geiser, etc.)
>
> File Changes
>
>   * M helm-shell.el (47)
>
> Patch Links:
>
>   * https://github.com/emacs-helm/helm/pull/2133.patch
>   * https://github.com/emacs-helm/helm/pull/2133.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.*


-- 
Thierry

**Comment by Ambrevar** [comment]
All done.

File: helm-shell.el [hunk 1]
```diff
@@ -133,10 +137,10 @@ See `helm-shell-prompts-list'."
 (defun helm-shell-prompts ()
   "Pre-configured `helm' to browse the prompts of the current Shell."
   (interactive)
-  (if (eq major-mode 'shell-mode)
+  (if (cl-member-if 'derived-mode-p helm-shell-comint-mode-list)
```
Comment (thierryvolpiatto) [inline helm-shell.el]: You can use `(apply 'derived-mode-p helm-shell-comint-mode-list)`, which is simpler, same below .

**Comment by thierryvolpiatto** [comment]
Thanks, I have resolved conflict and replaced `cl-member-if` with `apply`.

**Comment by Ambrevar** [comment]
Never noticed that `derived-mode-p` took `&rest` arguments... Thanks for pointing it out!

**Pull request merged**